runtime/browser/xwalk_notification_manager_linux.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_BROWSER_XWALK_NOTIFICATION_MANAGER_LINUX_H_
+#define XWALK_RUNTIME_BROWSER_XWALK_NOTIFICATION_MANAGER_LINUX_H_
+
+#include <libnotify/notification.h>
+#include <libnotify/notify.h>
+
+#include <map>
+
+#include "base/callback.h"
+#include "base/containers/scoped_ptr_hash_map.h"
+#include "base/strings/string16.h"
+
+class GURL;
+class SkBitmap;
+
+namespace content {
+class BrowserContext;
+class DesktopNotificationDelegate;
+class RenderFrameHost;
+struct PlatformNotificationData;
+}
+
+namespace xwalk {
+
+class XWalkNotificationManagerLinux {
+ public:
+  XWalkNotificationManagerLinux();
+  virtual ~XWalkNotificationManagerLinux();
+
+  // Show a desktop notification. If |cancel_callback| is non-null, it's set to
+  // a callback which can be used to cancel the notification.
+  void ShowDesktopNotification(
+      content::BrowserContext* browser_context,
+      const GURL& origin,
+      const content::PlatformNotificationData& notification_data,
+      scoped_ptr<content::DesktopNotificationDelegate> delegate,
+      int render_process_id,
+      base::Closure* cancel_callback);
+
+  void NotificationDisplayed(NotifyNotification* notification);
+  void NotificationClicked(NotifyNotification* notification);
+  void NotificationClosed(NotifyNotification* notification, bool by_user);
+
+  gulong closed_handler(NotifyNotification* notification) {
+    return notifications_handler_map_[notification];
```

Comment:
do not use '[]' operator here, it will implicitly create new entity for the given 'notification'